docs/samples/generic-host/BuilderApi/Program.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Threading.Tasks;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+class Program
+{
+    static Task<int> Main(string[] args)
+        => new HostBuilder()
+        .RunCommandLineApplicationFluentAsync(args, (app, provider) =>
```

Comment:
I'm not sure I understand the purpose of the `provider` argument. Can we change the required usage to this?

```suggestion
        .RunCommandLineApplicationAsync(args, app =>
```

Reply:
In [PR 350](https://github.com/natemcmaster/CommandLineUtils/pull/350), I needed the IServiceProvider instance for injecting dependencies while configuring the application (for example, IConsole and other application-specific dependencies)

Reply:
This just seems like an API that's hard to understand the first time you see it. I'd like to consider alternative ways to make the service provider available.

Reply:
Makes sense to me.  To be more detailed... if someone (like me) wants to take advantage of constructor injection by using [ActivatorUtilities.CreateInstance](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilities.createinstance?view=dotnet-plat-ext-3.1) in their configuration method, then they will need an instance of IServiceProvider.

Reply:
Maybe this is a little known fact, but `CommandLineApplication` implements `IServiceProvider`, so you should be able to calls `Activator.CreateInstance<T>(app)`. The only thing that should be required to make this work is for CommandLineSerivce to call `UseConstructorInjection`.

Reply:
Nice!  So, this code would allow constructor injection for the configure method?  ```(app) => ActivatorUtilities.CreateInstance<SomeClass>(app)```

Reply:
> I'm not sure I understand the purpose of the provider argument.

Unfortunately hadn't realized that `CommandLineApplication` implemented `IServiceProvider`. Thank you for explanation!

Reply:
After removing `IServiceProvider` arg my test for checking DI started to fail, because it couldn't resolve `IHostEnvironment` from the `CommandLineApplication` service provider. 
The problem was in missing call of `UseConstructorInjection` method in `CommandLineService` class. I didn't use this method in the builder api version because its summary stated `Enables using constructor injection to initialize the model type`. 
After reading the code of `ConstructorInjectionConvention` class I realized that aside from it's main responsibility it allowed the app to use `IServiceProvider` provided by the host. Maybe, we should split this convention in two parts or give it a more general name?

Reply:
> Maybe, we should split this convention in two parts or give it a more general name?

Probably. Let's handle that in a separate issue or pull-request.